### PR TITLE
Update developer guide on implementation of update commands

### LINKED ISSUE #173: Add update commands implementation in DG

### PR DESCRIPTION
**Pull request by lirc572**
Closes #173 

Preview: <https://lirc572.github.io/tp/DeveloperGuide.html#update-commands-coming-soon-in-v13>

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -206,6 +206,56 @@ Step 4. The `Ui` calls `UiCommand#execute(MainWindow)`, which will result in a c
     * Cons: 
         * `Ui` and `CommandResult` are not closed to modification. A new instruction might require the addition of fields to `CommandResult` (to store instructions and related data) as well as a new conditional statement in `Ui` to handle the new instruction.
 
+### Update Commands [Coming soon in v1.3]
+
+CoLAB has several update commands for projects, events, deadlines, tasks and groupmates. They are used to edit details of entities that have already been created.
+
+The update project command and the update contact command are relatively straight forward. Below is a sequence diagram of how an update projecct command is executed.
```
Comment (samuelfangjw) [inline docs/DeveloperGuide.md]: "The update project command and the update contact command are relatively straight forward."

This line doesn't really add much value, can consider removing

Comment (samuelfangjw) [inline docs/DeveloperGuide.md]: typo in projecct.

Also can consider styling it. `UpdateProject` command is executed

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -206,6 +206,56 @@ Step 4. The `Ui` calls `UiCommand#execute(MainWindow)`, which will result in a c
...
+
+Step 2. User input is passed to the `addressBookParser`, which creates a new `UpdateProjectCommand`.
+
+Step 3. The `UpdateProjectCommand` will then be executed by calling its `execute` method.
+
+Step 4. Since the `ModelManager` is passed to `UpdateProjectCommand#execute`, it is able to call a method `ModelManager#setProject` to change an existing project of a given `Index` in the `ProjectsFolder` to the modified version.
+
+Step 5. After the project gets updated, `Model#saveProjectsFolder` is called to save the list of projects to files.
+
+The other update commands require some more work because events, deadlines, tasks abd groupmates are sub-components of a project. It is therefore necessary to specify a project in the command so that edits can be applied to that project. Below is a sequence diagram of how an update groupmate command is executed.
```
Comment (samuelfangjw) [inline docs/DeveloperGuide.md]: typo "abd"

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -206,6 +206,56 @@ Step 4. The `Ui` calls `UiCommand#execute(MainWindow)`, which will result in a c
...
+
+##### Aspect: How the target contact is specified when updating contacts
+
+* **Alternative 1 (current choice):** Pass the `Index` object down to `UniquePersonList#setPerson`.
+    * Pros: More Consistent in how to pass indexes and locate an element in a `List` throughout the codebase.
+    * Cons: Higher coupling since `UniquePersonList` now relies on `Index`.
+
+* **Alternative 2 (implementation used in AB3):** Pass the target `Person` object to be edited to `UniquePersonList#setPerson`.
+    * Pros: Lower coupling since `Index` is not a dependency of `UniquePersonList`.
+    * Cons: We must ensure that the implementation of each individual command are correct.
```
Comment (samuelfangjw) [inline docs/DeveloperGuide.md]: Not very sure what is meant by "We must ensure that the implementation of each individual command are correct.", maybe can rephrase.

**Review by samuelfangjw — CHANGES_REQUESTED** [review]
For the 2 commands, i think it's better to focus on the low level view instead.
Screenshot 2021-03-25 at 12 12 55 PM

This sequence diagram above already shows the high level view of how user interacts to create command and save result. Would be better for diagram to focus on just the low level details (what happens during the execution of the edit  or update command and maybe how the model is updated)

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (lirc572) [reply docs/DeveloperGuide.md]: My bad, forgot to change this line after copying from the sample.

**Review by samuelfangjw — APPROVED** [review]
lgtm

**Pull request merged**